File: src/main/java/org/example/kalkulationsprogramm/domain/SpamModelStats.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Globale Statistiken für das Naive Bayes Spam-Modell.
 * Speichert Zähler wie total_spam und total_ham (Anzahl trainierter Dokumente).
 */
@Entity
@Table(name = "spam_model_stats")
@Getter
@Setter
@NoArgsConstructor
public class SpamModelStats {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 50)
    private String statKey;

    @Column(nullable = false)
    private long statValue = 0;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/SpamModelStats.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*

@Entity
@Table(name = "spam_model_stats")
open class SpamModelStats {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(nullable = false, unique = true, length = 50)
    open var statKey: String? = null

    @Column(nullable = false)
    open var statValue: Long = 0

}